### PR TITLE
KDSKey Resource

### PR DESCRIPTION
**Pull request by rdbartram**
When creating gMSAs a KDSRootKey is required. This is simply a resource to create one.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xactivedirectory/139)

### CODE REVIEW

File: DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.meta.mof [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[ClassVersion("1.0.0.0"), FriendlyName("xKDSKey")]
```
Comment (BerheAbrha) [inline DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.meta.mof]: Please document the change in the unreleased section in https://github.com/PowerShell/xActiveDirectory/blob/dev/README.md

File: DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.psm1 [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        [Parameter(Mandatory)]
+        [DateTime] $StartTime
+    )
+
+    try
+    {
+        $KDSKeys = @(Get-KDSRootKey)
+        return @{
+            AttributeOfWrongFormat = $KDSKeys[0].AttributeOfWrongFormat
+            KeyValue = $KDSKeys[0].KeyValue
```
Comment (BerheAbrha) [inline DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.psm1]: why is get returning property values not defined in the schema ?  #needsdesignreview

File: DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.meta.mof (hunk 1 above)
Comment (rdbartram) [inline DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.meta.mof]: Sorry I can do that. In other modules the readme is in root. I thought it was strange one didn't exist.

File: DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.psm1 (hunk 2 above)
Comment (rdbartram) [inline DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.psm1]: Sorry, I forgot to create read properties in the schema. I will make the changes and update today

File: DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.meta.mof (hunk 1 above)
Comment (rdbartram) [inline DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.meta.mof]: Update as requested

File: DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.psm1 (hunk 2 above)
Comment (rdbartram) [inline DSCResources/MSFT_xKDSKey/MSFT_xKDSKey.psm1]: Updated as requested

**Comment by bgelens** [comment]
@BerheAbrha Can you finish the review? Also, please don't accept the visual studio files that came with this pr

**Comment by rdbartram** [comment]
Closed as there was outstanding items in review that I couldn't close and item wasn't being pick up by maintainers. Created new PR #165 in order that it get accepted

**Pull request closed without merging**